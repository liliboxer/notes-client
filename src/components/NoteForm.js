import React from 'react';
import PropTypes from 'prop-types';

function NoteForm({ title, body, handleTitleChange, handleBodyChange, handleSubmit }) {
  return (
    <>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={title}
          placeholder="title"
          onChange={handleTitleChange}>
        </input>
        <textarea
          type="text"
          value={body}
          placeholder="body"
          onChange={handleBodyChange}>
        </textarea>
        <button>Create Note</button>
      </form>
    </>
  );
}

NoteForm.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  handleTitleChange: PropTypes.func.isRequired,
  handleBodyChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
};

export default NoteForm;
