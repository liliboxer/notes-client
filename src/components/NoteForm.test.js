import React from 'react';
import { shallow } from 'enzyme';
import NoteForm from '../components/NoteForm';

describe('NoteForm component', () => {
  it('renders NoteForm', () => {
    const wrapper = shallow(<NoteForm 
      title="My Note"
      body="My awesome note"
      handleTitleChange={() => {}}
      handleBodyChange={() => {}}
      handleSubmit={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
