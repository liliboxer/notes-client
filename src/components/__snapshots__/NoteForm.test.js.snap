// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteForm component renders NoteForm 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="title"
      type="text"
      value="My Note"
    />
    <textarea
      onChange={[Function]}
      placeholder="body"
      type="text"
      value="My awesome note"
    />
    <button>
      Create Note
    </button>
  </form>
</Fragment>
`;
